types/simulation/import.go

```diff
@@ -0,0 +1,74 @@
+	_ "cosmossdk.io/api/cosmos/auth/module/v1"
+	_ "cosmossdk.io/api/cosmos/auth/v1beta1"
+	_ "cosmossdk.io/api/cosmos/authz/module/v1"
+	_ "cosmossdk.io/api/cosmos/authz/v1beta1"
+	_ "cosmossdk.io/api/cosmos/autocli/v1"
+	_ "cosmossdk.io/api/cosmos/bank/module/v1"
+	_ "cosmossdk.io/api/cosmos/bank/v1beta1"
+	_ "cosmossdk.io/api/cosmos/base/abci/v1beta1"
+	_ "cosmossdk.io/api/cosmos/base/node/v1beta1"
+	_ "cosmossdk.io/api/cosmos/base/query/v1beta1"
+	_ "cosmossdk.io/api/cosmos/base/reflection/v1beta1"
+	_ "cosmossdk.io/api/cosmos/base/reflection/v2alpha1"
+	_ "cosmossdk.io/api/cosmos/base/tendermint/v1beta1"
+	_ "cosmossdk.io/api/cosmos/base/v1beta1"
+	_ "cosmossdk.io/api/cosmos/consensus/module/v1"
+	_ "cosmossdk.io/api/cosmos/consensus/v1"
+	_ "cosmossdk.io/api/cosmos/crisis/module/v1"
+	_ "cosmossdk.io/api/cosmos/crisis/v1beta1"
+	_ "cosmossdk.io/api/cosmos/crypto/ed25519"
+	_ "cosmossdk.io/api/cosmos/crypto/hd/v1"
+	_ "cosmossdk.io/api/cosmos/crypto/multisig"
+	_ "cosmossdk.io/api/cosmos/crypto/multisig/v1beta1"
+	_ "cosmossdk.io/api/cosmos/crypto/secp256k1"
+	_ "cosmossdk.io/api/cosmos/crypto/secp256r1"
+	_ "cosmossdk.io/api/cosmos/distribution/module/v1"
+	_ "cosmossdk.io/api/cosmos/distribution/v1beta1"
+	_ "cosmossdk.io/api/cosmos/evidence/module/v1"
+	_ "cosmossdk.io/api/cosmos/evidence/v1beta1"
+	_ "cosmossdk.io/api/cosmos/feegrant/module/v1"
+	_ "cosmossdk.io/api/cosmos/feegrant/v1beta1"
+	_ "cosmossdk.io/api/cosmos/genutil/module/v1"
+	_ "cosmossdk.io/api/cosmos/genutil/v1beta1"
+	_ "cosmossdk.io/api/cosmos/gov/module/v1"
+	_ "cosmossdk.io/api/cosmos/gov/v1"
+	_ "cosmossdk.io/api/cosmos/gov/v1beta1"
+	_ "cosmossdk.io/api/cosmos/group/module/v1"
+	_ "cosmossdk.io/api/cosmos/group/v1"
+	_ "cosmossdk.io/api/cosmos/mint/module/v1"
+	_ "cosmossdk.io/api/cosmos/mint/v1beta1"
+	_ "cosmossdk.io/api/cosmos/msg/textual/v1"
+	_ "cosmossdk.io/api/cosmos/msg/v1"
+	_ "cosmossdk.io/api/cosmos/nft/module/v1"
+	_ "cosmossdk.io/api/cosmos/nft/v1beta1"
+	_ "cosmossdk.io/api/cosmos/orm/module/v1alpha1"
+	_ "cosmossdk.io/api/cosmos/orm/query/v1alpha1"
+	_ "cosmossdk.io/api/cosmos/orm/v1"
+	_ "cosmossdk.io/api/cosmos/orm/v1alpha1"
+	_ "cosmossdk.io/api/cosmos/params/module/v1"
+	_ "cosmossdk.io/api/cosmos/params/v1beta1"
+	_ "cosmossdk.io/api/cosmos/query/v1"
+	_ "cosmossdk.io/api/cosmos/reflection/v1"
+	_ "cosmossdk.io/api/cosmos/slashing/module/v1"
+	_ "cosmossdk.io/api/cosmos/slashing/v1beta1"
+	_ "cosmossdk.io/api/cosmos/staking/module/v1"
+	_ "cosmossdk.io/api/cosmos/staking/v1beta1"
+	_ "cosmossdk.io/api/cosmos/store/snapshots/v1"
+	_ "cosmossdk.io/api/cosmos/store/streaming/abci"
+	_ "cosmossdk.io/api/cosmos/store/v1beta1"
+	_ "cosmossdk.io/api/cosmos/tx/config/v1"
+	_ "cosmossdk.io/api/cosmos/tx/signing/v1beta1"
+	_ "cosmossdk.io/api/cosmos/tx/v1beta1"
+	_ "cosmossdk.io/api/cosmos/upgrade/module/v1"
+	_ "cosmossdk.io/api/cosmos/upgrade/v1beta1"
+	_ "cosmossdk.io/api/cosmos/vesting/module/v1"
+	_ "cosmossdk.io/api/cosmos/vesting/v1beta1"
+)
```

Comment:
None of this should be necessary. This is only needed because the amino JSON encoder isn't properly setup to take a custom file resolver and use dynamicpb when there are is no concrete type

Reply:
I've moved the any resolving and type casting up into this sims package because I think in this case we can permit packed anys without an amino.name (aminojson.Encoder asserts this case).  But I still need a typeResolver to get a protoreflect enabled msg for aminojson.Encoder.  Any suggestions?

Reply:
Use dyncamicpb if the global typeresolver doesn't have the type like I was saying above. Also the encoder should be parameterized by both file resolver and type resolver. See what autocli does

Reply:
>the encoder should be parameterized by both file resolver and type resolver. See what autocli does

Can we tackle this in a separate PR?  I don't think any change is required here for sims to remove this last usage.

Reply:
I guess so but addressing this issue with aminojson takes priority in my opinion. We said that it would work dynamically without depending on the API module. This PR has uncovered that this is not true. So we need to address that

Reply:
We can merge both, I don't see them as blocking, but all right.